File: src/main/java/com/fatec/destino/repository/pacote/TransporteRepository.java
package com.fatec.destino.repository.pacote;

import com.fatec.destino.model.pacote.transporte.Transporte;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;

public interface TransporteRepository extends JpaRepository<Transporte, Long> {
    @Query("SELECT t.preco FROM Transporte t WHERE t.id = :id")
    int findPrecoPorId(@Param("id") long id);
}

File: src/main/kotlin/com/fatec/destino/repository/pacote/transporte/TransporteRepository.kt
package com.fatec.destino.repository.pacote.transporte

import com.fatec.destino.model.pacote.transporte.Transporte
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository

@Repository
interface TransporteRepository : JpaRepository<Transporte, Long> {
    @Query("SELECT t.preco FROM Transporte t WHERE t.id = :id")
    fun findPrecoPorId(@Param("id") id: Long): Int
}